src/pages/en/guides/migration-guide/from-nextjs.md

```diff
@@ -0,0 +1,771 @@
+
+1. **Keep** Next's `public` folder untouched. 
+   
+    Astro uses the `public` directory for static assets, just like Next. There is no change needed to this folder, nor its contents.
+
+2. **Copy or Move** Next's other files and folders (e.g. `pages`, `styles` etc.) into Astro's `src/` folder.
+
+    Like Next, Astro's `src/pages` folder is a special folder used for file-based routing. All other folders are optional, and you can organize the contents of your `src/` folder any way you like. Common other folders in Astro projects include `src/layouts/`, `src/components`, `src/styles`, `src/scripts`, and for file-based routing of blog posts, Markdown or MDX files are contained within `pages`, for example `src/pages/posts/` or `src/pages/blog/` depending on the desired URL.
+
+
+### Repurpose config file
+
+Astro, like Next, [has a configuration file at the root of your project called `astro.config.mjs`](/en/guides/configuring-astro/). This is used only for configuring your Astro project and any installed integrations, including SSR adapters.
+
+
+### Converting JSX files to `.astro` files
+
+Here are some common actions you will perform when you convert a Next `.js` component into a `.astro` component:
+
+1. Use the returned JSX of the existing Next.js component function as the basis for your HTML template
+2. Change any [Next or JSX syntax to Astro](#convert-syntax-to-astro) (e.g. `<Link>`, `<Script>`, `{children}`, `className`, inline style objects) or to HTML web standards.
+3. Move any necessary JavaScript, including import statements, into a "code fence" (`---`). This includes:
+   - Rewriting any Next function props to use `Astro.props`.
+   - Converting any imported components to Astro themselves, too.
+   - Migrating `getStaticProps` logic out of a function and into the code fence. 
+   - Migrate JavaScript used to conditionally render content inside the HTML template directly.
+
+#### Example: JSX to Astro
+
+Compare the following Next component and a corresponding Astro component:
+
+```jsx title="Page.jsx"
+import Header from "./header";
+import Footer from './footer';
+import "./layout.css";
+
+export async function getStaticProps() {
+    const res = await fetch('https://api.github.com/repos/withastro/astro')
+    const json = await res.json()
+    return {
+        props: {
+            message: json.message,
+            stars: json.stargazers_count || 0,
+        },
+    }
+}
+
+const Component = ({
+    stars,
+    message,
+}) => {
+    if (message === "Not Found") return (
+        <p>The repository you're looking up doesn't exist</p>
+    )
+
+    return (
+        <>
+            <Header />
+            <p style={{
+                backgroundColor: `#f4f4f4`,
+                padding: `1em 1.5em`,
+                textAlign: `center`,
+                marginBottom: `1em`
+            }}>Astro has {stars} 🧑‍🚀</p>
+            <Footer />
+        </>
+    )
+}
+
+export default Component;
+```
```

Comment:
The NextJS section is 100% not ready for review, but I wanted to get your early thoughts on this code sample @sarah11918 

I couldn't find anything that quite utilized the same amount of code principles we had in the Gatsby site. What's worse, Gatsby doesn't have a great single strong example; instead [they opt for many many smaller examples](https://nextjs.org/examples).

As a result, I took an existing example: https://github.com/vercel/next.js/blob/canary/examples/data-fetch/pages/preact-stars.tsx

And modified it for our use.

The modified version shows how you would migrate:

- Styling migrations
- Data fetching
- Conditional rendering

All in one semi-small code sample. What do you think? Do we like this or should I replace it with something else?

Reply:
Early returns look good! I think you've done a great job of finding an example that even somewhat mirrors what we were able to do with the Gatsby example. 

I *love* showing a conversion with `getStaticProps()`. I think with Gatsby, leaning into the GraphQL and `createPages` made sense. With Next.js, the main differences are going to be different, and certainly `getStaticProps()` *is* something that was on my radar... since `getStaticPaths()` is a shared feature, but the props, not so much.

Let's absolutely roll with this!